app/controllers/trainees/outcome_details_controller.rb

```diff
@@ -2,19 +2,24 @@
 
 module Trainees
   class OutcomeDetailsController < ApplicationController
+    before_action :authorize_trainee
+
     def confirm
-      authorize trainee
       @outcome = OutcomeDateForm.new(trainee)
     end
 
     def recommended
-      authorize trainee
+      ensure_trainee_is_not_draft!
```

Comment:
Again, this could be a policy and should be a stricter check. You should only be able to see this recommended page if `trainee.qts_awarded?` I think.